Gems/PythonAssetBuilder/Editor/Scripts/scene_api/physics_data.py

```diff
@@ -0,0 +1,580 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+# for underlying data structures, see Code\Framework\AzFramework\AzFramework\Physics\Shape.h
+
+class ColliderConfiguration():
+    """
+    Configuration for a collider
+
+    Attributes
+    ----------
+    Trigger: bool
+        Should this shape act as a trigger shape.
+
+    Simulated: bool
+        Should this shape partake in collision in the physical simulation.
+
+    InSceneQueries: bool
+        Should this shape partake in scene queries (ray casts, overlap tests, sweeps).
+
+    Exclusive: bool
+        Can this collider be shared between multiple bodies?
+
+    Position: [float, float, float] Vector3
+        Shape offset relative to the connected rigid body.
+
+    Rotation: [float, float, float, float] Quaternion
+        Shape rotation relative to the connected rigid body.
+
+    ColliderTag: str
+        Identification tag for the collider.
+
+    RestOffset: float
+        Bodies will come to rest separated by the sum of their rest offsets.
+
+    ContactOffset: float
+        Bodies will start to generate contacts when closer than the sum of their contact offsets.
+    Methods
+    -------
+    to_dict()
+        Converts contents to a Python dictionary
+    """
+    def __init__(self):
+        self.Trigger = True
+        self.Simulated = True
+        self.InSceneQueries = True
+        self.Exclusive = True
+        self.Position = [0.0, 0.0, 0.0]
+        self.Rotation = [0.0, 0.0, 0.0, 1.0]
+        self.ColliderTag = ''
+        self.RestOffset = 0.0
+        self.ContactOffset = 0.02
+
+    def to_dict(self):
+        return self.__dict__
+
+# for underlying data structures, see Code\Framework\AzFramework\AzFramework\Physics\Character.h
+
+class CharacterColliderNodeConfiguration():
+    """
+    Shapes to define the animation's to model of physics
+
+    Attributes
+    ----------
+    name : str
+        debug name of the node
+
+    shapes : `list` of `tuple` of (ColliderConfiguration, ShapeConfiguration) 
+        a list of pairs of collider and shape configuration
+
+    Methods
+    -------
+    add_collider_shape_pair(colliderConfiguration, shapeConfiguration)
```

Comment:
All these comment blocks in the class headers are great, it makes it easy to follow and understand what's going on.

I don't know enough about how Python handles documentation, would it make more sense to move the method documentation to the methods themselves, so the function names don't get duplicated?

Reply:
Numpydoc supports both, and normally the classes that I have developed are simple helper functions to organize data